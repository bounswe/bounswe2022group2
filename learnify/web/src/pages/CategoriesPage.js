import React from 'react';
import './style.css';
import Footer from '../components/Footer';
import NavBar from '../components/NavBar';
import Categories from '../components/Categories';
import illustration from '../images/categories-page-illustration.svg'

function CategoriesPage() {

    return(
        <div>
            <NavBar/>
            <div className='categories-page-content'>
                <div className='categories-banner'>
                    <div>
                        <div className='categories-heading'>Looking for some categories?</div>
                        <div className='categories-phrase'>
                            In Learnify we have tons of various learning spaces for every curiosity you have. Browse the selection of our categories below.
                        </div>
                    </div>
                    <div>
                    <img src={illustration} alt="Learnify Logo" height={400} />
                    </div>
                </div>
                <Categories/>
            </div>
            <Footer/>
        </div>
    )
}

export default CategoriesPage;