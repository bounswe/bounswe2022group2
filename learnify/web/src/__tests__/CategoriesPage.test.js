import React from "react";
import CategoriesPage from "./../pages/CategoriesPage";
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("CategoriesPage", () => {
    test('should render page layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><CategoriesPage /></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
});

test('categories page text renders correctly', async () => {
    render(<BrowserRouter><CategoriesPage /></BrowserRouter>);
    const heading = screen.getByText(/Looking for some categories?/i);
    expect(heading).toBeInTheDocument();
    const phrase = screen.getByText(/In Learnify we have tons of various learning spaces for every curiosity you have. Browse the selection of our categories below./i);
    expect(phrase).toBeInTheDocument();
}   );

test('categories illustration renders correctly', async () => {
    render(<BrowserRouter><CategoriesPage /></BrowserRouter>);
    const illustration = screen.getByAltText(/Categories Page Illustration/i);
    expect(illustration).toBeInTheDocument();
}   );

test('footer renders correctly', async () => {
    render(<BrowserRouter><CategoriesPage /></BrowserRouter>);
    const footer = screen.getByTestId(/footer/i);
    expect(footer).toBeInTheDocument();
}   );

test('navbar renders correctly', async () => {
    render(<BrowserRouter><CategoriesPage /></BrowserRouter>);
    const navbar = screen.getByTestId(/navbar/i);
    expect(navbar).toBeInTheDocument();
}   );