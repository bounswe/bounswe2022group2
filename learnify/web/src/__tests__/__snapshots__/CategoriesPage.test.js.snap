// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesPage should render page layout correctly 1`] = `
<div>
  <nav
    className="nav"
    data-testid="navbar"
  >
    <a
      href="/home"
    >
      <img
        alt="Learnify Logo Navbar"
        height={60}
        src="logo-dblue.png"
      />
    </a>
    <ul>
      <div
        className="relative"
      >
        <input
          className="search-input-field"
          onChange={[Function]}
          placeholder="Search learning spaces"
          size={30}
          type="text"
          value=""
        />
        <div
          className="search-results-container"
        />
      </div>
      <li>
        <a
          className="navBarText"
          href="/notifications"
        >
          <img
            alt="Notifications Icon"
            height={20}
            src="notification-icon.svg"
          />
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/categories"
        >
          Categories
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/contact"
        >
          Contact
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/profile/null"
        >
          Profile
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="categories-page-content"
  >
    <div
      className="categories-banner"
    >
      <div>
        <div
          className="categories-heading"
        >
          Looking for some categories?
        </div>
        <div
          className="categories-phrase"
        >
          In Learnify we have tons of various learning spaces for every curiosity you have. Browse the selection of our categories below.
        </div>
      </div>
      <div>
        <img
          alt="Categories Page Illustration"
          height={400}
          src="categories-page-illustration.svg"
        />
      </div>
    </div>
    <div
      className="category-wrapper"
    />
  </div>
  <div
    className="footer-parent-layout"
    data-testid="footer"
  >
    <div
      className="footer-container"
    >
      <div
        className="footer-row"
      >
        <div
          className="footer-column-l"
        >
          <div
            className="footer-item-bold"
          >
            Learnify
          </div>
          <div
            className="footer-item-reg"
          >
            is a learning space for everyone, where learning is 
            <br />
             achieved through collaboration.
          </div>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/home"
          >
            Home
          </a>
          <a
            className="footer-item-bold"
            href="/categories"
          >
            Categories
          </a>
          <a
            className="footer-item-bold"
            href="/notifications"
          >
            Notifications
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/profile"
          >
            Profile
          </a>
          <a
            className="footer-item-bold"
            href="/contact"
          >
            Contact
          </a>
          <a
            className="footer-item-bold"
            href="/about"
          >
            About
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/user/learning-spaces"
          >
            My Learning Spaces
          </a>
          <a
            className="footer-item-bold"
            href="/popular-learning-spaces"
          >
            Popular Learning Spaces
          </a>
          <a
            className="footer-item-bold"
            href="/recommended-learning-spaces"
          >
            Recommended Learning Spaces
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          2022
           Â© Learnify Team. All rights reserved.
        </div>
      </div>
    </div>
  </div>
</div>
`;
